src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs

```diff
@@ -191,6 +191,8 @@ internal async Task ApplySearchParameterStatus(IReadOnlyCollection<ResourceSearc
 
             _latestSearchParams = updatedSearchParameterStatus.Select(p => p.LastUpdated).Max();
 
+            _searchParameterStatusDataStore.SyncStatuses(updatedSearchParameterStatus);
```

Comment:
TODO: Currently investigating a potential timing issue that entails the C# dictionary being accessed before the search parameter info is synchronized.

Reply:
As discussed with @feordin, it makes sense to keep the single-instance dictionary update in the `UpsertStatuses()` method. This will ensure the dictionary for the calling instance is immediately updated, since there is a delay in propagating the notification to other instances. I added this back in in 101d872.